### PR TITLE
SQL: Fix catalog filtering in SYS COLUMNS

### LINKED ISSUE #40582: SQL: SYS COLUMNS doesn't consider catalog wildcard
`SYS COLUMNS` currently incorrectly considers catalog `%` as a filtering query when in fact it is not. This causes no columns to be returned which is incorrect.

### PR DESCRIPTION
**Pull request by costin**
Properly treat '%' as a wildcard for catalog filtering instead of doing
a straight string match.

Fix #40582

### CODE REVIEW

**Comment by costin** [comment]
I have updated the PR to fix also alias filtering which was not performed (only the index name was considered). Also added a test suite since mocking doesn't really work for this case.

File: x-pack/plugin/sql/qa/single-node/src/test/java/org/elasticsearch/xpack/sql/qa/single_node/JdbcCsvSpecIT.java [hunk 1]
```diff
@@ -5,10 +5,26 @@
...
+import static org.elasticsearch.xpack.sql.qa.jdbc.CsvTestUtils.specParser;
+
 public class JdbcCsvSpecIT extends CsvSpecTestCase {
+
+
+    @ParametersFactory(argumentFormatting = PARAM_FORMATTING)
+    public static List<Object[]> readScriptSpec() throws Exception {
+        List<Object[]> list = new ArrayList<>();
+        list.addAll(CsvSpecTestCase.readScriptSpec());
+        return readScriptSpec("/disabled/command-sys.csv-spec", specParser());
```
Comment (matriv) [inline x-pack/plugin/sql/qa/single-node/src/test/java/org/elasticsearch/xpack/sql/qa/single_node/JdbcCsvSpecIT.java]: I don't like `disabled` in the path here.

**Review by astefan — APPROVED** [review]
LGTM

**Review by matriv — APPROVED** [review]
LGTM

**Review by bpintea — APPROVED** [review]

**Review by matriv — APPROVED** [review]
LGTM

File: x-pack/plugin/sql/qa/src/main/resources/single-node-only/command-sys.csv-spec [hunk 2]
```diff
@@ -0,0 +1,120 @@
...
+x-pack_plugin_sql_qa_single-node_integTestCluster  |null           |test\_emp      |last_name.keyword |12             |KEYWORD        |32766          |2147483647     |null           |null           |1              |null           |null           |12             |0               |2147483647       |10              |YES            |null           |null           |null           |null            |NO              |NO                
+x-pack_plugin_sql_qa_single-node_integTestCluster  |null           |test\_emp      |salary            |4              |INTEGER        |11             |4              |null           |10             |1              |null           |null           |4              |0               |null             |11              |YES            |null           |null           |null           |null            |NO              |NO                
+;
+
+sysColumnsWithTableLikeNoEscape
+SYS COLUMNS TABLE LIKE 'test_emp';
+
+// since there's no escaping test_emp means test*emp which matches also test_alias_emp
```
Comment (astefan) [inline x-pack/plugin/sql/qa/src/main/resources/single-node-only/command-sys.csv-spec]: `test_emp` should match `test`[one character]`emp` when there is no escape character, no?

**Review by astefan — COMMENTED** [review]
Added two comments, but am not 100% sure I am right about those.

**Comment by bpintea** [comment]
Gave the branch a try and if nothing was mistaken during the process:
* `sys columns catalog '%' table like 'employees'` (and `sys columns catalog '%' table like 'employees' like '%'`) returns an empty result set;
* `sys columns catalog 'distribution\_run'  table like '%'` will return a full list of all columns in all tables is the catalog (which is right); but `sys columns catalog 'distribution\_run'  table like ''` will return the same, except for the table name column which is empty (which seems wrong).

File: x-pack/plugin/sql/qa/src/main/resources/single-node-only/command-sys.csv-spec [hunk 3]
```diff
@@ -0,0 +1,120 @@
...
+---------------+---------------+---------------+--------------------+---------------+---------------+---------------+---------------+---------------+---------------+---------------+---------------+---------------+---------------+----------------+-----------------+----------------+---------------+---------------+---------------+---------------+----------------+----------------+------------------
+x-pack_plugin_sql_qa_single-node_integTestCluster  |null           |test\_emp      |birth_date        |93             |DATETIME       |24             |8              |null           |null           |1              |null           |null           |9              |3               |null             |1               |YES            |null           |null           |null           |null            |NO              |NO
```
[the remark this answers is not in the document]
Comment (costin) [reply x-pack/plugin/sql/qa/src/main/resources/single-node-only/command-sys.csv-spec]: This is a bug - the `test\_emp` is escaped so the name should be `test_emp`.

File: x-pack/plugin/sql/qa/src/main/resources/single-node-only/command-sys.csv-spec (hunk 2 above)
Comment (costin) [reply x-pack/plugin/sql/qa/src/main/resources/single-node-only/command-sys.csv-spec]: Unfortunately due to the current field caps features, we can't figure out the source index.

**Comment by bpintea** [comment]
Update after consulting with @costin:

> `sys columns catalog '%' table like 'employees'` (and `sys columns catalog '%' table like 'employees' like '%'`) returns an empty result set;

This (=empty result set) is correct, since the catalog argument [`cannot contain a string search pattern`](https://docs.microsoft.com/en-us/sql/odbc/reference/syntax/sqlcolumns-function).

> `sys columns catalog 'distribution\_run' table like ''` will return the same, except for the table name column which is empty (which seems wrong)

This is currently a known limitation; however, it should have no practical impact, since the applications ask for the columns of specific table.

**Review by bpintea — APPROVED** [review]

**Pull request merged**